### ISSUE TITLE
CI Failure k3s related [master branch]

### ISSUE THREAD

**Issue by dtomcej** [containous/maesh#460]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

Opened PR.

### What did you expect to see?

No CI Errors

### What did you see instead?

```

START: kubernetes_test.go:11: KubernetesSuite.SetUpSuite
--
884 | Starting k3s...
886 | time="2020-03-09T16:28:31Z" level=info msg="Created cluster network with ID 7ca5fb58ad1b018d45488bcb986a3b455058def3725b0949972adb706760abff"
887 | time="2020-03-09T16:28:31Z" level=info msg="Created docker volume  k3d-maesh-integration-images"
888 | time="2020-03-09T16:28:31Z" level=info msg="Creating cluster [maesh-integration]"
889 | time="2020-03-09T16:28:31Z" level=info msg="Creating server using docker.io/rancher/k3s:v0.10.1...\n"
890 | time="2020-03-09T16:28:31Z" level=info msg="Pulling image docker.io/rancher/k3s:v0.10.1...\n"
891 | time="2020-03-09T16:28:41Z" level=info msg="Booting 1 workers for cluster maesh-integration"
892 |  
892 | time="2020-03-09T16:28:42Z" level=info msg="SUCCESS: created cluster [maesh-integration]"
894 | time="2020-03-09T16:28:42Z" level=info msg="You can now use the cluster with:\n\nexport KUBECONFIG=\"$(k3d get-kubeconfig --name='maesh-integration')\"\nkubectl cluster-info"
894 |  
896 | Importing docker images in to k3s...
896 | Importing image: containous/maesh:latest
897 | Importing image: containous/whoami:v1.0.1
899 | Importing image: coredns/coredns:1.3.1
899 | Importing image: traefik:v2.1.1
900 | Creating kube client...
902 | Setting new kubeconfig path...
902 | Creating required namespaces...
903 | namespace/maesh created
904 |  
906 | namespace/test created
906 |  
907 | k3s start successfully.
908 | serviceaccount/coredns created
910 | clusterrole.rbac.authorization.k8s.io/system:coredns created
910 | clusterrolebinding.rbac.authorization.k8s.io/system:coredns created
912 | configmap/coredns created
912 | deployment.apps/coredns created
913 | service/kube-dns created
914 |  
916 | kubernetes_test.go:19:
916 | s.startAndWaitForCoreDNS(c)
917 | integration_test.go:276:
919 | c.Assert(s.try.WaitReadyDeployment("coredns", metav1.NamespaceSystem, 60*time.Second), checker.IsNil)
920 | ... value *errors.errorString = &errors.errorString{s:"unable get the deployment \"coredns\" in namespace \"kube-system\": deployment not ready"} ("unable get the deployment \"coredns\" in namespace \"kube-system\": deployment not ready")
920 |  
921 | FAIL: kubernetes_test.go:11: KubernetesSuite.SetUpSuite
922 |  
924 | START: kubernetes_test.go:28: KubernetesSuite.TestProviderConfig
925 | MISS: kubernetes_test.go:28: KubernetesSuite.TestProviderConfig
925 |  
926 | START: kubernetes_test.go:24: KubernetesSuite.TearDownSuite
928 | time="2020-03-09T16:32:44Z" level=info msg="Removing cluster [maesh-integration]"
929 | time="2020-03-09T16:32:44Z" level=info msg="...Removing 1 workers\n"
929 | time="2020-03-09T16:32:44Z" level=info msg="...Removing server"
931 | time="2020-03-09T16:32:45Z" level=info msg="...Removing docker image volume"
932 | time="2020-03-09T16:32:45Z" level=info msg="Removed cluster [maesh-integration]"
932 |  
934 | PASS: kubernetes_test.go:24: KubernetesSuite.TearDownSuite	0.912s
934 |  
935 | OOPS: 0 passed, 1 FAILED, 1 MISSED
936 | --- FAIL: Test (285.21s)


```

**Comment by dtomcej** [comment]
@jspdown/@mmatur do you have any suggestions here?

We currently install k3s with no coreOS, then manually install our own version.

Sometimes it is slow....like really slow......

I'm wondering if there are any ways to improve this step

**Comment by dtomcej** [comment]
Should be resolved by adding the readiness checks in #461
